src/bot/exts/dragonfly/dragonfly.py

```diff
@@ -261,6 +262,199 @@ async def report(self: Self, interaction: discord.Interaction, button: discord.u
+
+        self.embed.add_field(name="Notes", value="\n".join(notes), inline=False)
+
+        if event_log:
+            self.embed.add_field(name="Event log", value=event_log, inline=False)
+
+        await interaction.message.edit(embed=self.embed, view=self.view)
+
+    async def on_error(
+        self, interaction: discord.Interaction, error: Exception,
+    ) -> None:
+        """Handle errors that occur in the modal."""
+        await interaction.response.send_message(
+            "An unexpected error occured.", ephemeral=True,
+        )
+        raise error
+
+    @property
+    def interaction(self) -> discord.Interaction | None:
+        """Get the interaction that triggered the modal."""
+        return self._interaction
+
+
+class MalwareView(discord.ui.View):
+    """View for the malware triage system."""
+
+    message: discord.Message | None = None
+
+    def __init__(
+        self: Self, embed: discord.Embed, bot: Bot, payload: PackageScanResult,
+    ) -> None:
+        self.embed = embed
+        self.bot = bot
+        self.payload = payload
+        self.event_log = []
+
+        super().__init__()
+
+    async def enable_button(self, button_label: str) -> None:
+        """Enables a button by its label."""
+        for button in self.children:
+            if button.label == button_label:
+                button.disabled = False
+
+    async def add_event(self, message: str) -> None:
+        """Add an event to the event log."""
+        # Much like earlier, we need to check the fields of the embed to determine where to add the event log
+        match len(self.embed.fields):
+            case 0:
+                pass
+            case 1:
+                if self.embed.fields[0].name == "Event log":
+                    self.embed.clear_fields()
+            case 2:
+                if self.embed.fields[0].name == "Event log":
+                    self.embed.clear_fields()
+                elif self.embed.fields[1].name == "Event log":
+                    self.embed.remove_field(1)
+
+        self.event_log.append(
+            message,
+        )  # For future reference, we save the event log in a variable
+        self.embed.add_field(
+            name="Event log", value="\n".join(self.event_log), inline=False,
+        )
+
+    async def update_status(self, status: str) -> None:
+        """Update the status of the package in the embed."""
+        self.embed.set_footer(text=status)
+
+    def get_timestamp(
+        self,
+    ) -> (
+        int
+    ):
+        """Returns the current timestamp in seconds."""
+        return f"<t:{int(datetime.now(UTC).timestamp())}:R>"
+
+    @discord.ui.button(
+        label="Report",
+        style=discord.ButtonStyle.red,
+    )
+    async def report(
+        self, interaction: discord.Interaction, button: discord.ui.Button[MalwareView],
+    ) -> None:
+        """Report package and update the embed."""
+        await self.enable_button("Approve")
```

Comment:
I think we should use the button objects directly
```suggestion
        self.approve.disabled = False
```
This will also allow us to remove the `enable_button` function